core-bundle/src/Resources/contao/widgets/SerpPreview.php

```diff
@@ -70,7 +70,7 @@ public function generate()
 
 		if ($titleTag = $this->getTitleTag($model))
 		{
-			$title = StringUtil::substr(sprintf($titleTag, $title), 64);
+			$title = StringUtil::substr(sprintf(str_replace('%', '%%', $titleTag), $title), 64);
```

Comment:
Wouldn’t this stop the `sprintf()` from working for the expected `%s` placeholder?

Reply:
I think the correct place for this fix is in https://github.com/contao/contao/blob/465dc7457491fa9066806def5f73e6dbdfac2b99/core-bundle/src/Resources/contao/dca/tl_page.php#L1021 (and in the same method of the other DCAs)

Reply:
True, didn't think about this part of code enough.

Reply:
I am using `preg_replace('/%(?!s)/', '%%', $titleTag)` now, which should replace all occurrences of `%` with `%%`, except `%s`. wdyt?

Replacing the `%` with `%%` before `%s` gets added is otherwise a bit cumbersome, as it might look like this for example:

```php
// Replace page title with a placeholder and any other insert tags with its actual value
$strTitleTag = self::replaceInsertTags(str_replace('{{page::pageTitle}}', '[[PAGETITLE]]', $layout->titleTag ?: '{{page::pageTitle}} - {{page::rootPageTitle}}'));

// Escape any "%" characters and then replace page title placeholder with a sprintf placeholder
$strTitleTag = str_replace('[[PAGETITLE]]', '%s', str_replace('%', '%%', $strTitleTag));

return $strTitleTag;
```

Reply:
This would still break with `%s` in the page title. How about something like this?

```php
return implode(
	'%s', 
	array_map(
		function($strVal) {
			return str_replace('%', '%%', self::replaceInsertTags(strVal));
		},
		explode('{{page::pageTitle}}', $layout->titleTag ?: '{{page::pageTitle}} - {{page::rootPageTitle}}', 2)
	)
);
```